### PR TITLE
Adds an optional parameter to kerneldbgen to time-pad smithed kernels.

### PR DESCRIPTION
**Pull request by krlberry**
##  Description
Adds a new optional parameter to kerneldbgen (the ISIS application that generates kernel databases) to add time-padding to database entries for smithed kernels. This was added in addition to the global startoffset/endoffset parameters because for #3363, just the smithed kernel entries needed to have their start/stop times extended by a fraction of a second.

## Related Issue
#3363 

## Motivation and Context
Fixes issue where smithed kernels would not get picked up by spiceinit because of a fraction of a second time-offset from the start time of the image, even though they work with every pixel of an image. 

## How Has This Been Tested?
Updated `makedb` scripts for themis to use updated kerneldbgen, ran with 0.5 second padding added to the smithed themis SPKs and CKs, and now spiceinit will pick up the smithed kernels. 

Also used phocube and stats to make sure that pointing information was available for every pixel of the image.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/system/apps/kerneldbgen/kerneldbgen.xml [hunk 1]
```diff
@@ -429,7 +432,21 @@
           <item>SMITHEDFILTER</item>
         </exclusions>
       </parameter>
-    </group>
+      <parameter name="SMITHEDPAD">
+        <type>double</type>
+        <brief>
+          Seconds to subtract from the start time and add to the stop time in each smithed kernel.
```
Comment (jessemapel) [inline isis/src/system/apps/kerneldbgen/kerneldbgen.xml]: This is added to each segment of each kernel, not the entire start and stop time of the kernel.

**Review by jessemapel — COMMENTED** [review]
Looks good, but needs a changelog entry

File: isis/src/system/apps/kerneldbgen/kerneldbgen.xml (hunk 1 above)
Comment (krlberry) [reply isis/src/system/apps/kerneldbgen/kerneldbgen.xml]: Thanks for catching this. I'll update the documentation.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
@kberryUSGS This can be merged once CI successfully builds.

**Review by scsides — CHANGES_REQUESTED** [review]
Each smithed kernel filter pattern usually comes from a single source. I think it would be reasonable to have a different padding for each filter. It might also be reasonable to add the padding for reconstructed too

**Comment by krlberry** [comment]
This didn't actually end up being useful for the THEMIS kernel problem I was working on. @scsides Do you think it's still worth me updating this and getting it merged? I was considering just closing this PR because I'm not sure it's actually useful.

**Comment by scsides** [comment]
This would only be useful if there were different sources of smithed kernels that needed different padding. If we don't need it now, I suggest closing. We can always add the feature in later when it is needed.

**Comment by jessemapel** [comment]
@scsides The other time I've run into this, I had to do what Kristin is doing here. It was for Cassini ISS images that all had radically different exposure times. I ended up writing a script to pad for each individual image.

**Pull request closed without merging**